src/string_immutable.zig

```diff
@@ -4037,6 +4037,19 @@ pub fn encodeBytesToHex(destination: []u8, source: []const u8) usize {
     return to_read * 2;
 }
 
+/// Leave a single leading char
+/// ```zig
+/// trimSubsequentLeadingChars("foo\n\n\n\n", '\n') -> "foo\n"
+/// ```
+pub fn trimSubsequentLeadingChars(slice: []const u8, char: u8) []const u8 {
+    var end = slice.len - 1;
```

Comment:
integer underflow when slice.len is 0
```suggestion
    var end = slice.len -| 1;
```